powershap/shap_wrappers/shap_explainer.py

```diff
@@ -152,44 +152,58 @@ def explain(
                         np.arange(len(X)), test_size=val_size, random_state=i, stratify=stratify
                     )
 
-            X_train = X.iloc[np.sort(train_idx)]
-            X_val = X.iloc[np.sort(val_idx)]
+            X_train = X.iloc[np.sort(train_idx)].to_numpy()
+            X_val = X.iloc[np.sort(val_idx)].to_numpy()
             Y_train = y[np.sort(train_idx)]
             Y_val = y[np.sort(val_idx)]
 
             Shap_values = self._fit_get_shap(
-                X_train=X_train.values,
+                X_train=X_train,
                 Y_train=Y_train,
-                X_val=X_val.values,
+                X_val=X_val,
                 Y_val=Y_val,
-                random_seed=i + random_seed_start,
+                random_seed=random_seed,
                 **kwargs,
             )
 
+            # If our data is large, we don't want to hold two copies in memory
+            # at once, so we manually release them before the next assignment.
+            del X_train, X_val, Y_train, Y_val
+
             Shap_values = np.abs(Shap_values)
 
             if len(np.shape(Shap_values)) > 2:
                 Shap_values = np.max(Shap_values, axis=0)
 
             # TODO: consider to convert to even float16?
-            shaps += [np.mean(Shap_values, axis=0).astype("float32")]
+            Shap_values = np.mean(Shap_values, axis=0).astype("float32")
+
+            shaps += [Shap_values]
+
+            # Python's reference-counting garbage collection has a hard time
+            # cleaning up after some of the work above in at least some cases,
+            # so running a manual gc.collect() here helps a great deal.
+            gc.collect()
+
 
         shaps = np.array(shaps)
 
-        return pd.DataFrame(data=shaps, columns=X_train.columns.values)
+        Xcolumns = X.columns.values
+        X.drop(random_col_name, axis=1, inplace=True)
```

Comment:
Oh yes, that would be better. Will fix

Reply:
Ok, now I am realizing that maybe I don't need to worry about restoring the X data since I think it's always a copy... But I would also have to remove that assert line at the top of this function. I'll push up a commit with these lines removed, for your review.